pkg/cvo/cvo_test.go

```diff
@@ -1353,6 +1353,229 @@ func TestOperator_sync(t *testing.T) {
+						ClusterID: configv1.ClusterID(id),
+						Upstream:  configv1.URL("http://localhost:8080/graph"),
+						Channel:   "fast",
+					},
+					Status: configv1.ClusterVersionStatus{
+						History: []configv1.UpdateHistory{
+							{State: configv1.CompletedUpdate, Version: "0.0.1-abc", Image: "image/image:v4.0.1", StartedTime: defaultStartedTime, CompletionTime: &defaultCompletionTime},
+						},
+						Desired:            configv1.Update{Image: "image/image:v4.0.1", Version: "0.0.1-abc"},
+						ObservedGeneration: 2,
+						Conditions: []configv1.ClusterOperatorStatusCondition{
+							{Type: configv1.OperatorAvailable, Status: configv1.ConditionTrue, Message: "Done applying 0.0.1-abc"},
+							{Type: configv1.OperatorProgressing, Status: configv1.ConditionFalse, Message: "Cluster version is 0.0.1-abc"},
+							{Type: ClusterStatusFailing, Status: configv1.ConditionFalse},
+							{Type: configv1.RetrievedUpdates, Status: configv1.ConditionFalse},
+							{Type: configv1.ClusterStatusConditionType("Upgradeable"), Status: configv1.ConditionFalse},
+							{Type: configv1.ClusterStatusConditionType("UpgradeableA"), Status: configv1.ConditionFalse},
+							{Type: configv1.ClusterStatusConditionType("UpgradeableB"), Status: configv1.ConditionFalse},
+						},
+					},
+				})
+			},
+		},
+		{
+			name: "new upgradable conditions with some old ones, version is live and was recently synced, sync",
+			syncStatus: &SyncWorkerStatus{
+				Generation:  2,
+				Reconciling: true,
+				Completed:   1,
+				Actual:      configv1.Update{Image: "image/image:v4.0.1", Version: "0.0.1-abc"},
+			},
+			optr: Operator{
+				releaseImage: "image/image:v4.0.1",
+				namespace:    "test",
+				name:         "default",
+				upgradeable: &upgradeable{
+					Conditions: []configv1.ClusterOperatorStatusCondition{
+						{Type: configv1.ClusterStatusConditionType("Upgradeable"), Status: configv1.ConditionFalse},
+						{Type: configv1.ClusterStatusConditionType("UpgradeableB"), Status: configv1.ConditionFalse},
+					},
+				},
+				client: fakeClientsetWithUpdates(&configv1.ClusterVersion{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:       "default",
+						Generation: 2,
+					},
+					Spec: configv1.ClusterVersionSpec{
+						ClusterID: configv1.ClusterID(id),
+						Upstream:  configv1.URL("http://localhost:8080/graph"),
+						Channel:   "fast",
+					},
+					Status: configv1.ClusterVersionStatus{
+						ObservedGeneration: 2,
+						Conditions: []configv1.ClusterOperatorStatusCondition{
+							{Type: configv1.OperatorAvailable, Status: configv1.ConditionFalse},
+							{Type: configv1.OperatorProgressing, Status: configv1.ConditionFalse, Message: "Cluster version is 0.0.1-abc"},
+							{Type: ClusterStatusFailing, Status: configv1.ConditionFalse},
+							{Type: configv1.RetrievedUpdates, Status: configv1.ConditionFalse},
+							{Type: configv1.ClusterStatusConditionType("UpgradeableA"), Status: configv1.ConditionFalse},
```

Comment:
Fold this into a constant in pkg/cvo (like ClusterStatusFailing) or move into api (former is faster, can do the others after)

Reply:
Or just use OperatorUpgradeable (isn't that in api already?)

Reply:
> https://github.com/openshift/cluster-version-operator/pull/243/files#diff-745b27730a4db8ced457d29d6656b64aR1433

So this test is actually testing that the original clusterversion object has a condition `UpgradeableA` set, but this new sync needs `Upgradeable` and `UpgradeableB` (https://github.com/openshift/cluster-version-operator/pull/243/files#diff-745b27730a4db8ced457d29d6656b64aR1446-R1447) to be set.
and the end result was exactly that https://github.com/openshift/cluster-version-operator/pull/243/files#diff-745b27730a4db8ced457d29d6656b64aR1500-R1501